tools/autograd/gen_python_functions.py

```diff
@@ -515,7 +515,7 @@ def gen_has_torch_function_check(
     if noarg:
         if method:
             return f"""\
-if(check_has_torch_function(self_)) {{
+if(!THPVariable_CheckExact(self_) && check_has_torch_function(self_)) {{
```

Comment:
Why do you need this extra check here?

Reply:
This is a fast path. I've come to the conclusion that because I've written something very similar to `check_has_torch_function` in https://github.com/pytorch/pytorch/pull/48965, the logic really should be merged. Which probably means the check will be removed from this PR, and `check_torch_function` will be replaced later in the PR stack.